### PR TITLE
sc-5665 Implement beneficiary policy functions

### PR DESCRIPTION
**Pull request by pdeziel**
This rounds off the sync/async implementation work by implementing the policy handlers on the beneficiary and does some cleanup to try to reduce common code.

### CODE REVIEW

File: pkg/rvasp/db/db.go [hunk 1]
```diff
@@ -82,11 +86,65 @@ func (d *DB) LookupPending() *gorm.DB {
...
+	return d.Query().Where("envelope = ?", envelope)
+}
+
 // LookupWallet by wallet address.
 func (d *DB) LookupWallet(address string) *gorm.DB {
 	return d.Query().Where("address = ?", address)
 }
 
+// CreateTransaction returns a new Transaction from the originator and beneficiary
+// wallet addresses, ready to be modified and/or stored in the database.
```
Comment (pdeziel) [inline pkg/rvasp/db/db.go:100]: Rename or update docstring because create implies creating something in the database.

File: pkg/rvasp/db/db.go [hunk 2]
```diff
@@ -179,6 +238,7 @@ const (
 // protocol and the envelope ID that uniquely identifies the message chain.
 type Transaction struct {
 	gorm.Model
+	TxID          string           `gorm:"not null"`
```
Comment (pdeziel) [inline pkg/rvasp/db/db.go:241]: Add marshaled version of the transaction payload like we're doing with the identity payload.

File: pkg/rvasp/transfer.go [hunk 3]
```diff
@@ -77,6 +77,28 @@ func createTransferPayload(identity *ivms101.IdentityPayload, transaction *gener
...
+	payload = &protocol.Payload{
+		SentAt: time.Now().Format(time.RFC3339),
+	}
+
+	if pending == nil {
+		return nil, fmt.Errorf("nil pending message supplied")
+	}
+
+	if payload.Identity, err = anypb.New(&ivms101.IdentityPayload{}); err != nil {
+		return nil, fmt.Errorf("could not dump payload identity: %s", err)
```
Comment (pdeziel) [inline pkg/rvasp/transfer.go:91]: Identity should be passed in so that this returns a valid payload.

File: pkg/rvasp/rvasp.go [hunk 4]
```diff
@@ -187,15 +187,28 @@ func (s *Server) Transfer(ctx context.Context, req *pb.TransferRequest) (*pb.Tra
 	policy := wallet.Policy
```
Comment (pdeziel) [inline pkg/rvasp/rvasp.go:187]: Add debug message with the wallet and policy.

File: pkg/rvasp/rvasp.go [hunk 5]
```diff
@@ -187,15 +187,28 @@ func (s *Server) Transfer(ctx context.Context, req *pb.TransferRequest) (*pb.Tra
...
+		// transaction.
+		return s.initAsyncTransfer(req, account)
 	case db.RejectedAsync:
-		return s.asyncTransfer(req, account)
+		// The rejected async scenario starts an asynchronous transfer with the
+		// beneficiary by sending a transfer request containing the complete identity
+		// and a partial transaction. The beneficiary will eventually reject the
+		// transfer.
+		return s.initAsyncTransfer(req, account)
 	default:
```
Comment (pdeziel) [inline pkg/rvasp/rvasp.go:210]: Add log message if the policy is unknown.

**Review by DanielSollis — APPROVED** [review]
approved based on PR meeting

**Pull request merged**